### DISCUSSION TITLE
Drawing shapes to a Texture2D

### DISCUSSION THREAD

**Discussion by Chris-Mingay** [raysan5/raylib#4245] [Q&A]
Hi everyone, I like using shapes to generate in game sprites etc. However as drawing textures is much cheaper than drawing shapes I'd like to "bake" them to a texture and then just draw said texture.

Is that possible with Raylib?

Cheers!

**Comment by Chris-Mingay** [comment]
Fantastic, that's exactly what I'm after thank you

**Comment by CrackedPixel** [comment]
For something like this a good approach is to use a RenderTexture. You would create the render texture at whatever atlas resolution you want, then draw your shapes etc onto it once and only once. After that, you would use the render texture as a texture atlas and draw your sprites. 

There are several examples showing how to use RTs, here is one example:  https://www.raylib.com/examples/textures/loader.html?name=textures_mouse_painting
